### ISSUE TITLE
During high usage the base URL used for an API call dropped the path part

### ISSUE THREAD

**Issue by felixsand** [starweb/shop-api-sdk-php#16]
When using the SDK for thousands of requests the path part of the base URL was suddenly dropped, resulting in a lot of 404 errors.

I always used the same SDK calls:
```
$productsResponse = $client->listProducts(['filterSku' => $sku, 'include' => 'primaryVariant']);
```
This line was in a loop for ~4000 SKU's.
Most request called the URL:
`https://velltra-se.starwebserver.se/api/v2/products?filterSku={{SKU}}&include=primaryVariant`

but some unexpectedly used:
`https://velltra-se.starwebserver.se/products?filterSku={{SKU}}&include=primaryVariant`
instead.

With other words the base URL path `/api/v2` unexpectedly disappeared.

**Comment by felixsand** [comment]
When running 500 API calls a piece I cannot reproduce. You need at least >2000 it seam. Maybe this is linked to the API Rate limit somehow?

One approach to try to reproduce this locally might be to lower the rate limit to 10 or something like that.

**Comment by felixsand** [comment]
From @lordrhodos 
---
>indeed, seems to be some issue in the BaseUriPlugin which combines the AddHostPlugin and AddPathPlugin (edited) 
>there is an identifier collision which prevents the AddPathPlugin to alter the request
>will look a little bit further, maybe it relates here:

https://github.com/php-http/client-common/issues/118
https://github.com/php-http/client-common/issues/171

>seems like it will not be fixed for the 1.x branch so I will a) look into getting the sdk ready to use the httplug 2.x branch or b) come up with an own implementation of the plugin and use that in the sdk (edited)

**Comment by lordrhodos** [comment]
fixed by #19
